src/model/letter.rs

```diff
@@ -0,0 +1,47 @@
+use combine::{choice, token, Parser, Stream};
+use strum_macros::{Display, EnumIter};
+
+#[derive(Clone, Copy, Debug, Display, EnumIter, Eq, PartialEq)]
```

Comment:
`EnumIter` here generates a `Letter::iter()` function to get all the values of the enum.

Reply:
Interesting, I was reading about how `.iter()`, while "idiomatic", isn't actually a trait implementation so I guess this is just following that convention rather than implementing some common trait?

Reply:
Yes, I think so.